File: benchmarks/live_suite/fusion.rs
fn run_fusion(n: usize) -> i64 {
    let mut a = Vec::with_capacity(n);
    let mut b = Vec::with_capacity(n);
    let mut d = Vec::with_capacity(n);
    let mut c = vec![0i64; n];

    for i in 0..n {
        a.push(((i % 50) + 1) as i64);
        b.push(((i % 30) + 1) as i64);
        d.push(100i64);
    }

    for j in 0..n {
        let val = a[j] * b[j] - d[j];
        c[j] = if val > 0 { val } else { 0 };
    }

    c[n / 2]
}

fn main() {
    let mut res = 0;
    for _ in 0..10 {
        res = run_fusion(1000000);
    }
    println!("Fusion Checksum: {}", res);
}

File: bench/live_suite/fusion.go
package main

import "fmt"

func runFusion(n int) int64 {
	a := make([]int64, n)
	b := make([]int64, n)
	d := make([]int64, n)
	c := make([]int64, n)

	for i := 0; i < n; i++ {
		a[i] = int64((i % 50) + 1)
		b[i] = int64((i % 30) + 1)
		d[i] = 100
	}

	for j := 0; j < n; j++ {
		val := a[j]*b[j] - d[j]
		if val > 0 {
			c[j] = val
		} else {
			c[j] = 0
		}
	}

	return c[n/2]
}

func main() {
	var res int64 = 0
	for iter := 0; iter < 10; iter++ {
		res = runFusion(1000000)
	}
	fmt.Printf("Fusion Checksum: %d\n", res)
}
